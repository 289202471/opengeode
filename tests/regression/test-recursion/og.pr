SYSTEM og;
    /* CIF TEXT (159, 221), (287, 188) */
    -- Text area for declarations and comments

use dv comment 'dataview-uniq.asn';

signal dd(Tree);
signal we;
    /* CIF ENDTEXT */
    CHANNEL c
        FROM ENV TO og WITH dd;
        FROM og TO ENV WITH we;
    ENDCHANNEL;
    BLOCK og;
        SIGNALROUTE r
            FROM ENV TO og WITH dd;
            FROM og TO ENV WITH we;
        CONNECT c AND r;
        /* CIF PROCESS (225, 50), (150, 75) */
        PROCESS og;
            /* CIF TEXT (57, 58), (287, 140) */
            -- Text area for declarations and comments

dcl my_tree Tree;

dcl res path;
            /* CIF ENDTEXT */
            /* CIF PROCEDURE (460, 209), (75, 35) */
            PROCEDURE findpath;
                /* CIF TEXT (0, 131), (272, 140) */
                dcl result Path := { cost 0, elem {} };

dcl elem Path_elem := {};

dcl single Letter;
                /* CIF ENDTEXT */
                /* CIF TEXT (129, 18), (183, 98) */
                fpar
    in startnode letter,
    in endnode letter,
    in my_tree Tree,
    in/out best_path Path;
                /* CIF ENDTEXT */
                /* CIF START (521, 64), (70, 35) */
                START;
                    /* CIF TASK (503, 119), (106, 68) */
                    TASK best_path := {
    cost 100,
    elem {}
};
                    /* CIF DECISION (480, 207), (151, 50) */
                    DECISION startnode = endnode;
                        /* CIF ANSWER (264, 277), (70, 23) */
                        (true):
                            /* CIF TASK (246, 320), (106, 68) */
                            TASK best_path := {
    cost 0,
    elem {}
};
                            /* CIF RETURN (281, 408), (35, 35) */
                            RETURN ;
                        /* CIF ANSWER (566, 277), (70, 23) */
                        (false):
                            /* CIF DECISION (504, 320), (193, 50) */
                            DECISION length(my_tree(startnode))
= 0;
                                /* CIF ANSWER (362, 390), (70, 23) */
                                (true):
                                    /* CIF RETURN (379, 433), (35, 35) */
                                    RETURN ;
                                /* CIF ANSWER (664, 390), (70, 23) */
                                (false):
                                    /* CIF TASK (442, 433), (514, 113) */
                                    TASK for each in my_tree(startnode):
    call findpath(each!son, endnode, my_tree, result);
    task result!cost := result!cost + each!cost;
    task single := each!son,  elem := {single} // elem ;
    task best_path := if result!cost < best_path!cost then result else best_path fi;
endfor;
                                    /* CIF TASK (628, 566), (141, 35) */
                                    TASK result!elem := elem;
                                    /* CIF RETURN (681, 616), (35, 35) */
                                    RETURN ;
                            ENDDECISION;
                    ENDDECISION;
            ENDPROCEDURE;
            /* CIF START (155, 216), (70, 35) */
            START;
                /* CIF TASK (1, 271), (378, 128) */
                TASK my_tree(a) := {{ cost 3, son b}, {cost 4, son c}},
my_tree(b) := {{cost 2, son d}},
my_tree(c) := {{ cost 1, son d}, {cost 1, son e}, {cost 6, son f}},
my_tree(d) := {{cost 1, son h}},
my_tree(e) := {{cost 7, son h}},
my_tree(f) := {{cost 1, son g}},
my_tree(g) := {},
my_tree(h) := {{cost 5, son g}};
                /* CIF PROCEDURECALL (97, 419), (186, 35) */
                CALL findpath(a, g, my_tree, res);
                /* CIF PROCEDURECALL (89, 474), (201, 35) */
                CALL writeln('best cost:', res!cost);
                /* CIF NEXTSTATE (155, 524), (70, 35) */
                NEXTSTATE wait;
            /* CIF STATE (296, 231), (70, 35) */
            STATE wait;
            ENDSTATE;
        ENDPROCESS og;
    ENDBLOCK;
ENDSYSTEM;