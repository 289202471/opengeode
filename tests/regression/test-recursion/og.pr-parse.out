[INFO] Starting to parse model test-recursion/og.pr
[ERROR] Failed to parse model.
    org.antlr.v4.runtime.misc.ParseCancellationException: line 3:4 no viable alternative at input '/* CIF TEXT (313, 221), (287, 188) */\n    -- Text area for declarations and comments\n\nuse dv comment 'dataview-uniq.asn';\n\nsignal dd(Tree);\nsignal we;\n    /* CIF ENDTEXT */\n    CHANNEL c\n        FROM ENV TO og WITH dd;\n        FROM og TO ENV WITH we;\n    ENDCHANNEL;\n    BLOCK og;\n        SIGNALROUTE r\n            FROM ENV TO og WITH dd;\n            FROM og TO ENV WITH we;\n        CONNECT c AND r;\n        /* CIF PROCESS (308, 80), (138, 75) */\n        PROCESS og;\n            /* CIF TEXT (57, 58), (287, 140) */\n            -- Text area for declarations and comments\n\ndcl my_tree Tree;\n\ndcl res path;\n            /* CIF ENDTEXT */\n            /* CIF PROCEDURE (574, 209), (75, 35) */\n            PROCEDURE findpath;\n                /* CIF TEXT (107, 140), (329, 53) */\n                dcl result, best_path Path := { cost 0, elem {} };\n\ndcl elem Path_elem := {};\n                /* CIF ENDTEXT */\n                /* CIF TEXT (104, 47), (183, 68) */\n                fpar\n    in startnode letter,\n    in endnode letter;\nreturns Path;\n                /* CIF ENDTEXT */\n                /* CIF START (626, 64), (70, 35) */\n                START;\n                    /* CIF TASK (584, 119), (153, 23) */\n                    TASK best_path.cost := 100;\n                    /* CIF DECISION (585, 162), (151, 50) */\n                    DECISION startnode = endnode;\n                        /* CIF ANSWER (431, 232), (70, 23) */\n                        (true):\n                            /* CIF TASK (398, 275), (136, 23) */\n                            TASK best_path.cost :=0;\n                        /* CIF ANSWER (820, 232), (70, 23) */\n                        (false):\n                            /* CIF DECISION (742, 275), (226, 50) */\n                            DECISION length(my_tree(num(startnode)))\n= 0;\n                                /* CIF ANSWER (547, 345), (70, 23) */\n                                (true):\n                                /* CIF ANSWER (918, 345), (70, 23) */\n                                (false):\n                                    /* CIF TASK (708, 388), (490, 128) */\n                                    TASK for each in my_tree(num(startnode)):\n    result := findpath(each.son, endnode)\n    elem := result.elem\n    result.cost := result.cost + each.cost\n    elem := { each.son } // elem\n    result.elem := elem\n    best_path := if result.cost < best_path.cost then result else best_path fi\nendfor;\n                            ENDDECISION;\n                    ENDDECISION;\n                    /* CIF RETURN (643, 546), (35, 35) */\n                    RETURN best_path;\n            ENDPROCEDURE;\n            /* CIF START (237, 232), (70, 35) */\n            START;\n                /* CIF TASK (60, 287), (424, 128) */\n                TASK my_tree(Letter_a) := {{ cost 3, son b}, {cost 4, son c}},\nmy_tree(Letter_b) := {{cost 2, son d}},\nmy_tree(Letter_c) := {{ cost 1, son d}, {cost 1, son e}, {cost 6, son f}},\nmy_tree(Letter_d) := {{cost 1, son h}},\nmy_tree(Letter_e) := {{cost 7, son h}},\nmy_tree(Letter_f) := {{cost 1, son g}},\nmy_tree(Letter_g) := {},\nmy_tree(Letter_h) := {{cost 5, son g}};\n                /* CIF TASK (202, 435), (139, 35) */\n                TASK res := findpath(a, g);\n                /* CIF PROCEDURECALL (172, 485), (200, 35) */\n                CALL writeln('best cost:', res.cost);\n                /* CIF PROCEDURECALL (199, 540), (145, 35) */\n                CALL write('best path: [ ');\n                /* CIF TASK (196, 590), (151, 53) */\n                TASK for each in res.elem:\n    call write(each, ' ');\nendfor;\n                /* CIF PROCEDURECALL (211, 663), (122, 35) */\n                CALL writeln(']');\n                /* CIF NEXTSTATE (237, 713), (70, 35) */\n                NEXTSTATE wait;\n            /* CIF STATE (494, 231), (70, 35) */\n            STATE wait;\n            ENDSTATE;\n        ENDPROCESS og;\n    ENDBLOCK;\n'
org.antlr.v4.runtime.misc.ParseCancellationException: line 3:4 no viable alternative at input '/* CIF TEXT (313, 221), (287, 188) */\n    -- Text area for declarations and comments\n\nuse dv comment 'dataview-uniq.asn';\n\nsignal dd(Tree);\nsignal we;\n    /* CIF ENDTEXT */\n    CHANNEL c\n        FROM ENV TO og WITH dd;\n        FROM og TO ENV WITH we;\n    ENDCHANNEL;\n    BLOCK og;\n        SIGNALROUTE r\n            FROM ENV TO og WITH dd;\n            FROM og TO ENV WITH we;\n        CONNECT c AND r;\n        /* CIF PROCESS (308, 80), (138, 75) */\n        PROCESS og;\n            /* CIF TEXT (57, 58), (287, 140) */\n            -- Text area for declarations and comments\n\ndcl my_tree Tree;\n\ndcl res path;\n            /* CIF ENDTEXT */\n            /* CIF PROCEDURE (574, 209), (75, 35) */\n            PROCEDURE findpath;\n                /* CIF TEXT (107, 140), (329, 53) */\n                dcl result, best_path Path := { cost 0, elem {} };\n\ndcl elem Path_elem := {};\n                /* CIF ENDTEXT */\n                /* CIF TEXT (104, 47), (183, 68) */\n                fpar\n    in startnode letter,\n    in endnode letter;\nreturns Path;\n                /* CIF ENDTEXT */\n                /* CIF START (626, 64), (70, 35) */\n                START;\n                    /* CIF TASK (584, 119), (153, 23) */\n                    TASK best_path.cost := 100;\n                    /* CIF DECISION (585, 162), (151, 50) */\n                    DECISION startnode = endnode;\n                        /* CIF ANSWER (431, 232), (70, 23) */\n                        (true):\n                            /* CIF TASK (398, 275), (136, 23) */\n                            TASK best_path.cost :=0;\n                        /* CIF ANSWER (820, 232), (70, 23) */\n                        (false):\n                            /* CIF DECISION (742, 275), (226, 50) */\n                            DECISION length(my_tree(num(startnode)))\n= 0;\n                                /* CIF ANSWER (547, 345), (70, 23) */\n                                (true):\n                                /* CIF ANSWER (918, 345), (70, 23) */\n                                (false):\n                                    /* CIF TASK (708, 388), (490, 128) */\n                                    TASK for each in my_tree(num(startnode)):\n    result := findpath(each.son, endnode)\n    elem := result.elem\n    result.cost := result.cost + each.cost\n    elem := { each.son } // elem\n    result.elem := elem\n    best_path := if result.cost < best_path.cost then result else best_path fi\nendfor;\n                            ENDDECISION;\n                    ENDDECISION;\n                    /* CIF RETURN (643, 546), (35, 35) */\n                    RETURN best_path;\n            ENDPROCEDURE;\n            /* CIF START (237, 232), (70, 35) */\n            START;\n                /* CIF TASK (60, 287), (424, 128) */\n                TASK my_tree(Letter_a) := {{ cost 3, son b}, {cost 4, son c}},\nmy_tree(Letter_b) := {{cost 2, son d}},\nmy_tree(Letter_c) := {{ cost 1, son d}, {cost 1, son e}, {cost 6, son f}},\nmy_tree(Letter_d) := {{cost 1, son h}},\nmy_tree(Letter_e) := {{cost 7, son h}},\nmy_tree(Letter_f) := {{cost 1, son g}},\nmy_tree(Letter_g) := {},\nmy_tree(Letter_h) := {{cost 5, son g}};\n                /* CIF TASK (202, 435), (139, 35) */\n                TASK res := findpath(a, g);\n                /* CIF PROCEDURECALL (172, 485), (200, 35) */\n                CALL writeln('best cost:', res.cost);\n                /* CIF PROCEDURECALL (199, 540), (145, 35) */\n                CALL write('best path: [ ');\n                /* CIF TASK (196, 590), (151, 53) */\n                TASK for each in res.elem:\n    call write(each, ' ');\nendfor;\n                /* CIF PROCEDURECALL (211, 663), (122, 35) */\n                CALL writeln(']');\n                /* CIF NEXTSTATE (237, 713), (70, 35) */\n                NEXTSTATE wait;\n            /* CIF STATE (494, 231), (70, 35) */\n            STATE wait;\n            ENDSTATE;\n        ENDPROCESS og;\n    ENDBLOCK;\n'
	at ee.ibk.sdl.parser.ThrowingErrorListener.syntaxError(ThrowingErrorListener.java:18)
	at org.antlr.v4.runtime.ProxyErrorListener.syntaxError(ProxyErrorListener.java:41)
	at org.antlr.v4.runtime.Parser.notifyErrorListeners(Parser.java:544)
	at org.antlr.v4.runtime.DefaultErrorStrategy.reportNoViableAlternative(DefaultErrorStrategy.java:282)
	at org.antlr.v4.runtime.DefaultErrorStrategy.reportError(DefaultErrorStrategy.java:121)
	at ee.ibk.sdl.parser.grammar.SDLParser.entity_in_system(SDLParser.java:617)
	at ee.ibk.sdl.parser.grammar.SDLParser.system_definition(SDLParser.java:383)
	at ee.ibk.sdl.parser.grammar.SDLParser.pr_file(SDLParser.java:303)
	at ee.ibk.sdl.parser.SDLParserMain.doParse(SDLParserMain.java:40)
	at ee.ibk.sdl.parser.SDLParserMain.parseFile(SDLParserMain.java:72)
	at ee.ibk.sdl.parser.SDLParserMain.main(SDLParserMain.java:117)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.eclipse.jdt.internal.jarinjarloader.JarRsrcLoader.main(JarRsrcLoader.java:58)
